src/scanner.cc

```diff
@@ -248,7 +248,14 @@ struct Scanner {
             next();
           } else {
             stop();
-            return true;
+            next();
+            std::regex variable("[a-zA-Z_\x80-\xff]");
```

Comment:
Let's add a function that checks if a character is a valid identifier character rather than use a regex here.

The regex library doesn't seem super intuitive to use (imo), and I suspect it might be slightly less performant than a character range comparison. Performance probably isn't a big concern here but using a regex also doesn't seem to offer much value over a function with character comparisons.

Also, I think Tree-sitter actual converts the regexes used in grammar rules into character range comparisons. You'll find something like the snippet below in `src/parser.c`.

```c
case 86:
  ACCEPT_TOKEN(sym_identifier);
  if (('0' <= lookahead && lookahead <= '9') ||
      ('A' <= lookahead && lookahead <= 'Z') ||
      lookahead == '_' ||
      ('a' <= lookahead && lookahead <= 'z') ||
      (128 <= lookahead && lookahead <= 255)) ADVANCE(86);
  END_STATE();
```

Hacklang's own lexer defines valid identifier characters with [similar logic](https://github.com/facebook/hhvm/blob/fab9abb6b527ca3f3576148ef89e1ecb20bff95d/hphp/hack/src/parser/core/lexer.rs#L322-L328):

```rust
fn is_name_letter(c: char) -> bool {
    (c == '_')
        || ('0' <= c && c <= '9')
        || ('a' <= c && c <= 'z')
        || ('A' <= c && c <= 'Z')
        || ('\x7f' <= c)
}
```

Reply:
Good idea! I'll make these changes.